core/src/main/java/com/linecorp/armeria/common/util/CompositeException.java

```diff
@@ -155,12 +171,19 @@ public synchronized Throwable getCause() { // NOPMD
                             aggregateMessage.append(separator);
                         }
 
-                        for (int i = 0; i < depth + 2; i++) {
-                            aggregateMessage.append("  ");
-                        }
                         final StackTraceElement[] st = inner.getStackTrace();
                         if (st.length > 0) {
-                            aggregateMessage.append("at ").append(st[0]).append(separator);
+                            final boolean isEnableVerboseExceptionFlag =
+                                    verboseExceptionFlag.isSampled(inner.getClass());
+                            final int loopCount = isEnableVerboseExceptionFlag ?
+                                                  st.length : Math.min(
+                                                          disabledVerboseExceptionLoopCount, st.length);
+                            for (int i = 0; i < loopCount; i++) {
+                                for (int j = 0; j < depth + 2; j++) {
+                                    aggregateMessage.append("  ");
+                                }
```

Comment:
Should we pre-calculate the message padding and reuse it?
It would be inefficient to loop and append the same message.
```java
final String messagePadding = Strings.repeat("  ", depth + 2);
...

aggregateMessage.append(messagePadding).append("at ").append(st[i]).append(separator);
```

Reply:
I agree with that opinion. Iteration logic (calculate message padding **depth+2**) is being executed throughout the getCause() method.
Since there is no section in which the depth variable increases in the middle of the logic, i will finish the message padding calculation in advance.